src/main/kotlin/com/neva/gradle/fork/config/Config.kt

```diff
@@ -31,9 +29,11 @@ abstract class Config(private val fork: ForkExtension, val name: String) {
   val definedProperties: List<Property>
     get() {
       val others = mutableMapOf<String, Property>()
+      val context = PropertyContext(others)
```

Comment:
Don't like this cyclic reference. What about orchestrating fields toggles from outside?
```
fork {
    properties {
          definitions {
                 ...
          }
          dependencies {
                 property("enableAem").toggles("aemUrl", "aemLicencePath", ...)
          }
    }
}
```
Then you can extreact dependencies from properties and control dependencies from outside after each property value change.

Reply:
that approach is ok but mine is more elastic. I am following a rule "i will never know how users will use my plugin" so that generic hooks are ok. too strict API introduce limitations which I don't want to have